Count recently assigned bugs

ricky-rav opened the pull request:
In order to improve balance across team members when assigning new bugs, show also for each developer the number of bugs assigned in the past 21 days (moving window, symbolically equivalent to one sprint). Backports generated by Prow bot are excluded from the count.

The recently-assigned count requires n new queries to the JIRA server, with n = # of developers in the team. In order to skip this costly evaluation, the new command line argument "-q" or "--quick" should be used.

Improved output when a query to JIRA server fails: now a single dot is printed instead of the full error, which we wouldn't be able to fix on our side anyway.

Signed-off-by: Riccardo Ravaioli <rravaiol@redhat.com>

Commit:
Count recently assigned bugs

In order to improve balance across team members when assigning new bugs, show also for each developer the number of bugs assigned in the past 21 days (moving window, symbolically equivalent to one sprint).
Backports generated by Prow bot are excluded from the count.

The recently-assigned count requires n new queries to the JIRA server, with n = # of developers in the team. In order to skip this costly evaluation, the new command line argument "-q" or "--quick" should be used.

Improved output when a query to JIRA server fails: now a single dot is printed instead of the full error, which we wouldn't be able to fix on our side anyway.

Signed-off-by: Riccardo Ravaioli <rravaiol@redhat.com>

flavio-fernandes commented on network_bugs_overview:
```diff
@@ -317,17 +360,65 @@ def process_bz_bugs(bugs, developers):
+            latest_assign = assignee_changes[-1].created
+        else:
+            # if there are no assign changelogs, assume that the ticket was created
+            # already with the current assignee and consider the creatione date
```
super NIT: `creation`

flavio-fernandes commented:
cool beans!!!
/lgtm, short of a really silly nit

ricky-rav commented:
I've addressed your comment and replaced "recently" with "<=21 days" in the table printed by script:

```
Rank of developers least overloaded (today) [2023-01-24]:
========================================================
  #  Developer              Points    Bugs    NEW    ASSIGNED    POST       Assigned
                                                                           <=21 days
---  -------------------  --------  ------  -----  ----------  ------  -------------
  1  npinaeva@redhat.com        69       5      4           1       0              6
  2  jcaamano@redhat.com        99      10      6           4       0              3
  3  jtanenba@redhat.com       106       6      2           1       3              3
  4  bpickard@redhat.com       148      13     10           0       3              1
  5  mcambria@redhat.com       162       4      3           1       0              1
  6  jluhrsen@redhat.com       193       6      2           1       3              0
  7  mkennell@redhat.com       193      13      7           3       3              3
  8  rravaiol@redhat.com       198      10      7           3       0              3
  9  mmahmoud@redhat.com       215       7      4           3       0              9
 10  pepalani@redhat.com       217       5      1           2       2              2
 11  apanatto@redhat.com       236       5      1           2       2              2
 12  surya@redhat.com          325      10      4           5       1              0
 13  pdiak@redhat.com          344      13      3           5       5              9
 14  ffernand@redhat.com       476       7      3           2       2              8

```

Thanks, @flavio-fernandes! :)
